lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -176,6 +176,135 @@ class ConvertAtenBincountOp : public OpConversionPattern<AtenBincountOp> {
 };
 } // namespace
 
+namespace {
+class ConvertPseudoAtenIndexPutImplOp
+    : public OpConversionPattern<PseudoAtenIndexPutImplOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(PseudoAtenIndexPutImplOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    MLIRContext *context = op->getContext();
+    Value input = adaptor.self();
+    Value values = adaptor.values();
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType valuesType = values.getType().cast<RankedTensorType>();
+    Type resultElemType = typeConverter->convertType(op->getResult(0).getType())
+                              .cast<RankedTensorType>()
+                              .getElementType();
+
+    // TODO: Add support for the input with rank other than one.
+    if (inputType.getRank() != 1)
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: input rank other than one is not supported");
+
+    // The unsafe should be either `False` or `none`.
+    if (!op.unsafe().getType().isa<Torch::NoneType>()) {
+      bool unsafe;
+      if (!matchPattern(op.unsafe(), m_TorchConstantBool(&unsafe)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: unsafe must be a constant");
+      else if (unsafe)
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: unsafe is expected to be false");
+    }
+
+    // The accumulate should be a torch constant of boolean type.
+    bool accumulate;
+    if (!matchPattern(op.accumulate(), m_TorchConstantBool(&accumulate)))
+      return rewriter.notifyMatchFailure(
+          op, "Expected accumulate to be constant bool.");
+
+    // The element type of the `input` and `values` should be same.
+    if (inputType.getElementType() != valuesType.getElementType())
+      return rewriter.notifyMatchFailure(
+          op, "Input element type should be same as the values element type.");
+
+    SmallVector<Value> indicesList;
+    getListConstructElements(adaptor.indices(), indicesList);
+    // The size of the list of the index tensors should not be greater than the
+    // input rank.
+    if ((int64_t)indicesList.size() > inputType.getRank() &&
+        indicesList.size() > 0)
+      return rewriter.notifyMatchFailure(
+          op, "Indices list size should not be greater than the input rank.");
+
+    // TODO: Add support for cases with indices list size smaller than the input
+    // rank and indices list of size zero.
+    if ((int64_t)indicesList.size() < inputType.getRank() ||
+        indicesList.size() == 0)
```

Comment:
`indicesList.size() == 0` can be removed as it's included in the previous condition.